### DISCUSSION TITLE
Updating offers in bulk

### DISCUSSION THREAD

**Discussion by dominikahr** [allegro/allegro-api#8515] [Suggestions - resolved]
Hi there, 

Kaufland just launched a new endpoint which allows you to update up to 150 offers in one request and each offer in the request can have different data (for example price). It would be great if this was also implemeted to Allegro as right now we can edit up to 1000 offers in a bulk but only with the same values. So we could only update the price of 1000 products, if all the products would have a new same price.
![image](https://github.com/allegro/allegro-api/assets/147524238/57e2c0ad-0c98-4c75-82b4-cb621106e745)
https://sellerapi.kaufland.com/?page=inventory#updating-units-in-bulk

It is quite difficult to not go over the limit of 9000 requests per minute with so many clients that need to update product data on a regular basis. The only option that Allegro has for bulk editing is, that all the offers we update would need to have the sama data, which they mostly do not have, so we need to update each offer separately.

Would be great if you could think about a similar option for Allegro :)

Thanks in advance!

**Comment by DanielCyran** [comment]
Popieram ten wątek w 100%.

Dla integratorów obsługujących duże wolumeny ofert jest to absolutna podstawa,
a nie „nice to have”.
Obecne batch update w Allegro API, które wymagają identycznych wartości
dla wszystkich ofert, w praktyce nie rozwiązują problemu skali.

Minimum, które powinno zostać wprowadzone:

1. Endpoint do masowej aktualizacji stanów magazynowych (batch),
   w którym każda oferta w jednym requestcie może mieć inną wartość.

2. Osobny endpoint do masowej aktualizacji cen (batch),
   również z możliwością różnych wartości dla każdej oferty w jednym żądaniu.

Dodatkowo batch powinien umożliwiać aktualizację
co najmniej do 1000 ofert w jednym requestcie.

Takie podejście drastycznie ogranicza liczbę requestów,
zmniejsza ryzyko przekraczania limitów API
i realnie poprawia stabilność integracji przy wielu klientach.

Dziś, przy różnych cenach i stanach, integratorzy są zmuszeni
wysyłać aktualizacje praktycznie per oferta,
co przy większej skali jest nieefektywne i trudne do utrzymania.

Masowe aktualizacje z różnymi wartościami w jednym batchu
powinny być standardem w nowoczesnym API.

**Reply by PrzemyslawLukanowski** [reply]
Proszę jeszcze o chwilę cierpliwości - już wkrótce udostępnimy funkcjonalności, które spełnią opisane potrzeby.

**Reply by PrzemyslawLukanowski** [reply]
Dzień dobry, wczoraj udostępniliśmy nowy endpoint. Szczegóły znajdziecie w naszym [newsie](https://github.com/allegro/allegro-api/issues/13179).

**Reply by DanielCyran** [reply]
@PrzemyslawLukanowski 
Dziękujemy za udostępnienie nowych endpointów. W przyszłym tygodniu przeanalizujemy je i rozpoczniemy wdrażanie zmian po naszej stronie.

**Comment by kontakt-electrobit** [comment]
Dzień dobry
Czy jest już jakiś przybliżony termin wprowadzenia zmiany?

**Reply by PrzemyslawLukanowski** [reply]
Dzień dobry,  zasób powinien pojawić się na produkcji już wkrótce - nie później niż w ciągu najbliższych dwóch tygodni. Oczywiście zamieścimy odpowiedni news na naszej stronie.

**Reply by PrzemyslawLukanowski** [reply]
Dzień dobry, wczoraj udostępniliśmy nowy endpoint. Szczegóły znajdziecie w naszym [newsie](https://github.com/allegro/allegro-api/issues/13179).

**Comment by PrzemyslawLukanowski** [comment]
Hi @dominikahr, yesterday, we introduced a new endpoint. You can find all the details in our [news](https://github.com/allegro/allegro-api/issues/13179).
